src/app/src/test/kotlin/org/imperial/mrc/hint/integration/ModelRunTests.kt

```diff
@@ -0,0 +1,29 @@
+package org.imperial.mrc.hint.integration
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.imperial.mrc.hint.models.ModelRunParameters
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import org.springframework.boot.test.web.client.postForEntity
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+
+class ModelRunTests : SecureIntegrationTests() {
+
+    private fun getJsonEntity(): HttpEntity<String> {
+        val params = ModelRunParameters(1,1,1,mapOf())
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+        val jsonString = ObjectMapper().writeValueAsString(params)
+        return HttpEntity(jsonString, headers)
+    }
+
+    @ParameterizedTest
+    @EnumSource(IsAuthorized::class)
+    fun `can run model`(isAuthorized: IsAuthorized) {
```

Comment:
Do we need a test for when this fails to start a model running?

Reply:
In general I've gone with our integration tests just testing the happy path for each endpoint. Which does mean we're not testing that errors from hintr are correctly formatted, but that should all be under test in hintr itself. We could test the error path here too as extra insurance, would that be helpful for you?

Reply:
No sorry that is good - we probably only need a test when front end part is in that errors are shown. You're right that we should be covering the error cases in hintr.